students/Ramkumar_Rajanbabu/Lesson03/mailroom_p1.py

```diff
@@ -0,0 +1,163 @@
+#Mailroom Part 1
+
+import sys
+
+donors_db = [["William Gates, III", 653772.32, 12.17],
+            ["Mark Zuckerberg", 1663.23, 4300.87, 10432.0],
+            ["Jeff Bezos", 877.33],
+            ["Paul Allen", 663.23, 43.87, 1.32],
+            ["Ramkumar Rajanbabu", 200.30, 50.10, 5.25]]
+
+donors = [name[0] for name in donors_db] #List as donor names
+
+prompt = "\n".join(("Menu: ",
+                    "Please choose from below options:",
+                    "1 - Send a Thank You",
+                    "2 - Create a Report",
+                    "3 - Quit",
+                    ">>> "))
+def main():
+    """Display menu with options to user.
+    
+    Args:
+        None
+    Returns:
+        None
+    
+    """
+    
+    while True:
+        response = input(prompt)
+        if response == "1": #Send a Thank You
+            send_a_thank_you()
+        elif response == "2": #Create a Report
+            create_a_report()
+        elif response == "3": #Quit
+            exit_program()
+        else:
+            print("Please enter a valid option! (1-3)\n")
+
+def send_a_thank_you():
+    """Show a list of donors and add a new donor based on user response.
+    
+    Args:
+        None
+    Returns:
+        None
+
+    """
+
+    response = str(input("Type in the full donor name (or 'list' to view a list of donor names): "))
+    if response == "list":
+        view_donor_list()
+        send_a_thank_you() # Reprompt to main menu
+    elif response in donors:
+        donation_amount(response)
+    else:
+        donors.append(response)
+        donors_db.append([response]) #response in list because it's a new donor entry
+        donation_amount(response)
+
+def create_a_report():
+    """Create a report (table with values).
+    
+    Args:
+        None
+    Returns:
+        None
+
+    """
+    title_string = "{:<25s} | {:s} | {:s} | {:s}".format("Donor Name", "Total Given", "Num Gifts", "Average Gift")
+    title_bar = len(title_string) * "-"    
+    
+    donor_name = []
+    total_given = []
+    num_gifts = []
+    avg_gift = []
+
+    for donor in donors_db:
+        name = donor[0]
+        donor_name.append(name)
+
+        amount = donor[1:]
+        total = sum(amount)
+        total_given.append(total)
+
+        gifts = len(amount)
+        num_gifts.append(gifts)
+
+        avg = total / gifts
+        avg_gift.append(avg)
+    
+    print()
+    print(title_string)
+    print(title_bar)
+    for index in range(len(donor_name)):
+        f_string = "{:<25s}  ${:>11.2f}  {:>10d}  ${:>12.2f}".format(donor_name[index], total_given[index], num_gifts[index], avg_gift[index])
+        print(f_string)
```

Comment:
this function is missing the sorted feature (please check requirements).


I would also recommend creating a list of your summary like:
```
for donor in donors_db:
    summary.append((name, sum(donations), len(donations), avg)
```
then loop through your summary to create report - will simplify the code and it will make more sense when using with `sorted` function.